src/index.js

```diff
@@ -1,21 +1,34 @@
 import React from 'react'
 import immutagen from 'immutagen'
 
-export default component => {
-  const generator = immutagen(component)
+const compose = ({ next, value }) => next
+  ? React.cloneElement(value, null, values => compose(next(values)))
+  : value
 
-  const compose = context => {
-    const value = context.value
-    return context.next
-      ? React.cloneElement(value, null, values => compose(context.next(values)))
-      : value
-  }
+export default Component => {
+  const original = Component.prototype.render
+  const displayName = `EpitathContainer(${Component.displayName || 'anonymous'})`
+
+  if (!original) {
+    const generator = immutagen(Component)
 
-  function Epitath(props) {
-    return compose(generator(props))
+    return Object.assign(function Epitath(props) {
+      return compose(generator(props))
+    }, { displayName })
   }
 
-  Epitath.displayName = `EpitathContainer(${component.displayName || 'anonymous'})`
+  Component.prototype.render = function render() {
+    if (!render.generator) {
```

Comment:
good one, can you add some comments here explaining why this is needed for any future beginners trying to learn from this code?